field.go

```diff
@@ -989,7 +1017,7 @@ func (f *Field) SetValue(columnID uint64, value int64) (changed bool, err error)
 	}
 
 	// Determine base value to store.
-	baseValue := uint64(value - bsig.Min)
+	baseValue := int64(value - bsig.Min)
```

Comment:
On thinking about this: I think this should possibly be happening up above. The maximum isn't the value furthest from zero, it's the value furthest from bsig.Min. I think.